pkg/utils/validator/clusterresourceoverride.go

```diff
@@ -0,0 +1,92 @@
+package validator
+
+import (
+	"context"
+	"fmt"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+
+	fleetv1alpha1 "go.goms.io/fleet/apis/placement/v1alpha1"
+)
+
+// ValidateClusterResourceOverride validates cluster resource override fields and returns error.
+func ValidateClusterResourceOverride(ctx context.Context, client client.Client, operation admissionv1.Operation, cro fleetv1alpha1.ClusterResourceOverride) error {
+	allErr := make([]error, 0)
+
+	// Check if the resource is being selected by resource name
+	if err := ValidateResourceSelectedByName(cro); err != nil {
+		allErr = append(allErr, err)
+	}
+	// Get the list of cluster resource overrides
+	croList, err := getClusterResourceOverrideList(ctx, client)
+	if err != nil {
+		allErr = append(allErr, err)
```

Comment:
if we cannot get the list, we cannot validate the following so we can return directly.